documentation/high-avail/AMD-Pensando_HA_Proposal.md

```diff
@@ -288,7 +289,189 @@ typedef struct _sai_dash_ha_dpu_control_message_notification_data_t {
+        _In_ sai_object_id_t *dash_ha_vip_id,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list);
+
+/**
+ * @brief Policy Result Enumeration
+ */
+typedef enum _sai_dash_ha_policy_result_t {
+     /**
+      * @brief Default result - undefined
+      */
+     SAI_DASH_HA_POLICY_RESULT_NONE,
+
+     /**
+      * @brief Permit flow
+      */
+     SAI_DASH_HA_POLICY_RESULT_ALLOW,
+
+     /**
+      * @brief Deny flow
+      */
+     SAI_DASH_HA_POLICY_RESULT_DENY,
+
+} _sai_dash_ha_policy_result_t;
+
+/**
+ * @brief Bitmap values for HA Rewrite flags
+ */
+typedef enum _sai_dash_ha_rewrite_flags_t {
+
+    /**
+     * @brief Rewrite Initiator Flow DMAC
+     */
+    SAI_DASH_HA_REWRITE_IFLOW_DMAC = 1 << 0;
+
+    /**
+     * @brief Rewrite Initiator Flow Source IP Address
+     */
+    SAI_DASH_HA_REWRITE_IFLOW_SIP = 1 << 1;
+
+    /**
+     * @brief Rewrite Initiator Flow Source Port
+     */
+    SAI_DASH_HA_REWRITE_IFLOW_SPORT = 1 << 2;
+
+    /**
+     * @brief Rewrite Initiator Flow VNI
+     */
+    SAI_DASH_HA_REWRITE_IFLOW_VNI = 1 << 3;
+
+    /**
+     * @brief Rewrite Reverse Flow Source IP addresss
+     */
+    SAI_DASH_HA_REWRITE_RFLOW_SIP = 1 << 4;
+
+    /**
+     * @brief Rewrite Reverse Flow Destination IP Address
+     */
+    SAI_DASH_HA_REWRITE_RFLOW_DIP = 1 << 5;
+
+    /**
+     * @brief Rewrite Reverse Flow L4 Destination Port
+     */
+    SAI_DASH_HA_REWRITE_RFLOW_DPORT = 1 << 6;
+
+    /**
+     * @brief Rewrite Reverse Flow L4 Source Port
+     */
+    SAI_DASH_HA_REWRITE_RFLOW_SPORT = 1 << 7;
+
+    /**
+     * @brief Rewrite Reverse Flow VNI
+     */
+    SAI_DASH_HA_REWRITE_RFLOW_VNI = 1 << 8;
+
+} sai_dash_ha_rewrite_flags_t;
+
+/**
+ * @brief Rewrite info for Flow sync messages
+ *
+ */
+typedef struct _sai_dash_ha_rewrite_info_t {
+    /**
+     * @brief Initiator Flow DMAC
+     */
+    sai_mac_t iflow_dmac;
+
+    /**
+     * @brief Initiator Flow Source IP address
+     */
+    sai_ip_address_t iflow_sip;
+
+    /**
+     * @brief Initiator Flow L4 Source Port
+     */
+    sai_uint16_t iflow_sport;
+
+    /**
+     * @brief Initiator Flow VNID
+     */
+    sai_uint32_t iflow_vni;
+
+    /**
+     * @brief Initiator Flow Rewrite Flags expressed as a bit map of sai_dash_ha_rewrite_flags_t
+     */
+    sai_uint16_t iflow_flags;
+
+    /**
+     * @brief Reverse Flow Source IP address
+     */
+    sai_ip_address_t rflow_sip;
+
+    /**
+     * @brief Reverse Flow Destination IP address
+     */
+    sai_ip_address_t rflow_dip;
+
+    /**
+     * @brief Reverse Flow Destination Port
+     */
+    sai_uint16_t rflow_dport;
+
+    /**
+     * @brief Reverse Flow Source Port
+     */
+    sai_uint16_t rflow_sport;
+
+    /**
+     * @brief Reverse Flow VNID
+     */
+    sai_uint32_t iflow_vni;
+
+    /**
+     * @brief Reverse Flow Rewrite Flags expressed as a bit map of sai_dash_ha_rewrite_flags_t
+     */
+    sai_uint16_t iflow_flags;
```

Comment:
is 16 bits enough, should be keep it as 32 or 64 bits for future extension, btw, we should probably move this flag to the top.

Reply:
Agree. increased to a 64 bit value to be safe.